File: backend/app/schemas/patient.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class PatientBase(BaseModel):
    first_name: str
    last_name: str
    phone: str
    date_of_birth: datetime | None = None
    is_high_risk: bool = False
    address: str | None = None
    notes: str | None = None


class PatientCreate(PatientBase):
    pass


class PatientUpdate(BaseModel):
    first_name: str | None = None
    last_name: str | None = None
    phone: str | None = None
    date_of_birth: datetime | None = None
    is_high_risk: bool | None = None
    address: str | None = None
    notes: str | None = None


class PatientResponse(PatientBase):
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/patient.ts
export interface Patient {
  id: string;
  first_name: string;
  last_name: string;
  phone: string;
  date_of_birth: string | null;
  is_high_risk: boolean;
  address: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface PatientCreatePayload {
  first_name: string;
  last_name: string;
  phone: string;
  date_of_birth?: string | null;
  is_high_risk?: boolean;
  address?: string | null;
  notes?: string | null;
}

export interface PatientUpdatePayload {
  first_name?: string | null;
  last_name?: string | null;
  phone?: string | null;
  date_of_birth?: string | null;
  is_high_risk?: boolean | null;
  address?: string | null;
  notes?: string | null;
}
